doc/releases/migration-guide-4.1.rst

```diff
@@ -170,6 +170,29 @@ Pin Control
 
 Sensors
 =======
+  * The asynchronous sensor API now supports external clock sources. To use an external clock source
+    with the asynchronous sensor API, the following configurations are required:
+
+    * Enable one of the Kconfig options:
+      :kconfig:option:`CONFIG_SENSOR_CLOCK_COUNTER`,
+      :kconfig:option:`CONFIG_SENSOR_CLOCK_RTC`, or
+      :kconfig:option:`CONFIG_SENSOR_CLOCK_SYSTEM`.
+
+    * Define the ``zephyr,sensor-clock`` property in the device tree to specify the external clock source.
```

Comment:
This should be marked as "if not using the system clock". This node  isn't needed for the system timer.

Reply:
Done